### PR TITLE
zqd listen: Reformat startup logs

### LINKED ISSUE #1947: zqd listen: Starting log line db
Adjust the starting log line of the zqd listen command to include information about the db used.

As db is not applicable to workers and recruiters, this shouldn't appear in the log for these personalities.

### PR DESCRIPTION
**Pull request by mattnibs**
Create a named logger for manager, db and core and log when each of these
services has started, along with configuration.

Most important here is the "Connected" log line for the db logger, which prints out a password/username redacted version of the connection url used.

Closes #1947

### CODE REVIEW

**Comment by mattnibs** [comment]
This is what the start up logs will now look like:

```
$ zqd -db.kind=postgres -db.postgres.url='postgres://test:test@localhost:5432/test?sslmode=disable'
{"level":"info","ts":1610461684.4514098,"msg":"Open files limit raised","limit":10240}
{"level":"info","ts":1610461684.4759932,"logger":"db","msg":"Connected","kind":"postgres","uri":"postgres://t****:*****@localhost:5432/test?sslmode=disable"}
{"level":"info","ts":1610461684.477346,"logger":"manager","msg":"Loaded","root":"file:///Users/nibs/Code/zq"}
{"level":"info","ts":1610461684.4775512,"logger":"core","msg":"Started","personality":"all","suricata_supported":false,"zeek_supported":false}
{"level":"info","ts":1610461684.477679,"logger":"httpd","msg":"Listening","addr":"[::]:9867"}
```

File: ppl/zqd/apiserver/manager.go [hunk 1]
```diff
@@ -39,15 +39,10 @@ type Manager struct {
...
-	db, err := db.Open(ctx, logger, dbconf, root)
-	if err != nil {
-		return nil, err
-	}
-
+func NewManager(ctx context.Context, logger *zap.Logger, registerer prometheus.Registerer, root iosrc.URI, db db.DB) (*Manager, error) {
 	factory := promauto.With(registerer)
 	m := &Manager{
-		logger:   logger,
+		logger:   logger.Named("manager").With(zap.String("root", root.String())),
```
Comment (alfred-landrum) [inline ppl/zqd/apiserver/manager.go:45]: I don't think the root needs to be included on every manager log, how about adding it to the `Loaded` info line below once?

File: ppl/zqd/db/db.go [hunk 2]
```diff
@@ -35,11 +35,14 @@ type Config struct {
 }
 
 func Open(ctx context.Context, logger *zap.Logger, conf Config, root iosrc.URI) (DB, error) {
+	logger = logger.Named("db")
```
Comment (alfred-landrum) [inline ppl/zqd/db/db.go:38]: suggest using 'database' to make reading the logs a little easier.

```suggestion
	logger = logger.Named("database")
```

File: ppl/zqd/db/postgresdb/config.go [hunk 3]
```diff
@@ -46,3 +47,15 @@ func (c Config) String() string {
...
+
+// StringRedacted is the same as string except password and a username are
+// redacted. This should be used in logs.
+func (c Config) StringRedacted() string {
+	if c.Password != "" {
+		c.Password = strings.Repeat("*", 5)
+	}
+	if c.User != "" {
+		c.User = string(c.User[0]) + strings.Repeat("*", 4)
+	}
```
Comment (alfred-landrum) [inline ppl/zqd/db/postgresdb/config.go:59]: suggest always hiding them:

```suggestion
	c.Password = strings.Repeat("*", 5)
	c.User = strings.Repeat("*", 5)
```

File: ppl/zqd/db/postgresdb/postgresdb.go [hunk 4]
```diff
@@ -8,18 +8,25 @@ import (
...
 
 type PostgresDB struct {
-	db *pg.DB
+	db     *pg.DB
+	logger *zap.Logger
 }
 
-func Open(ctx context.Context, conf Config) (*PostgresDB, error) {
+func Open(ctx context.Context, logger *zap.Logger, conf Config) (*PostgresDB, error) {
+	logger = logger.With(
```
Comment (alfred-landrum) [inline ppl/zqd/db/postgresdb/postgresdb.go:20]: I don't think these need to be printed with each database log line; how about just adding them to the 'Connected' log below?

File: ppl/zqd/core.go [hunk 5]
```diff
@@ -109,25 +110,33 @@ func NewCore(ctx context.Context, conf Config) (*Core, error) {
...
 
+	personality := conf.Personality
+	if personality == "" {
+		personality = "all"
+	}
+
 	c := &Core{
 		auth:     auth,
-		logger:   conf.Logger,
+		logger:   conf.Logger.Named("core").With(zap.String("personality", personality)),
```
Comment (alfred-landrum) [inline ppl/zqd/core.go:120]: +1

File: ppl/zqd/apiserver/manager.go (hunk 1 above)
Comment (mattnibs) [reply ppl/zqd/apiserver/manager.go:45]: Sounds good.

**Review by alfred-landrum — APPROVED** [review]

**Pull request merged**